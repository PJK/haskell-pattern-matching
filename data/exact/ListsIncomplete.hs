module ListsIncomplete where

length :: [a] -> Int
length [_] = 1
length (x:xs) = 1 + (length xs)

