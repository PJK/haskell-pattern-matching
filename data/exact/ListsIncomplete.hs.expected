{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "length",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        [
                            {
                                "tag": "EmptyListPattern",
                                "contents": []
                            }
                        ],
                        ""
                    ]
                ]
            }
        ]
    ]
}
